### DISCUSSION TITLE
Workflow 3.0 - Can't get instance information after executing workflow

### DISCUSSION THREAD

**Discussion by vsanghavi-conga** [elsa-workflows/elsa-core#3730] [Q&A]
Hi,

I have setup Elsa 3.0 Designer and API using below links. I am able to Login and create workflow.
- https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-server
- https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-designer

But there are 2 issues we are facing.
1. There is not button on the UI to execute workflow.
2. I was able to execute workflow using Postman pointing to Execute API endpoint (/elsa/api/workflow-definitions/{definitionid}/execute) and it returns Instance Id of execution. But when I try to get Instance Information by calling Get Instance API endpoint (/elsa/api/workflow-instances/{instanceid}) I am getting Not Found.

Below is the screenshot of the workflow.
![image](https://user-images.githubusercontent.com/122509094/220825493-4c739675-a9f7-41e5-a10e-95cd34badea1.png)

Postman:
Execute Workflow:
![image](https://user-images.githubusercontent.com/122509094/220825713-453a213b-cc7e-4e91-ba47-c3efe2a75158.png)

Get Instance:
![image](https://user-images.githubusercontent.com/122509094/220825763-45249bec-c09b-476c-b148-e98e6b1c5061.png)

Regards,
Viraj

**Comment by sfmskywalker** [comment]
Hi @vsanghavi-conga

1. This feature has yet to be implemented.
2. Workflow instances are (currently) not persisted by default. To capture them we need to enable an exporter that captures & persists workflow instances. To do so, you can add the following code to the elsa setup delegate:

```csharp
services.AddElsa(elsa => elsa
        .UseWorkflowRuntime(runtime =>
        {
            // Configure the runtime provider to use the default workflow runtime and configure its persistence layer to use EF core using SQLite. Internally, this will also configure the workflow instance store to use EF Core.
            runtime.UseDefaultRuntime(dr => dr.UseEntityFrameworkCore(ef => ef.UseSqlite()));

            // Configure the runtime to persist execution log records using EF core using SQLite,
            runtime.UseExecutionLogRecords(e => e.UseEntityFrameworkCore(ef => ef.UseSqlite()));

            // Configure the runtime to export workflow state. The default handler will map the workflow state to the workflow instance model and persist it.
            runtime.UseAsyncWorkflowStateExporter();
        })
);
```

**Comment by vsanghavi-conga** [comment]
Hi @sfmskywalker , Thanks for the reply and solution, It worked and it's now giving result by Instance Id. Further I faced few more problems as well as have few questions.

1 I tried to setup workflow with If in designer but could not get idea about how to configure it. Is there any documentation on using designer to configure workflow?
![image](https://user-images.githubusercontent.com/122509094/221095045-935e7c7d-f1ea-432b-934d-5168bf1a91be.png)

2 When workflow execution runs into an error everything just gets stuck. After that I cannot even call API endpoint to execute a new instance.

3 When tried to view Instances from UI, page just greyed out and nothing was displayed.
![image](https://user-images.githubusercontent.com/122509094/221094291-9e5a04ff-8d3b-46f5-9197-2a493c4c7db8.png)

4 If we create a custom activity, will it be displayed automatically on the designer UI ? If not, What is required it to be displayed along with its parameters/configuration?

5 Is there any timeline to release Elsa 3.0 stable version ?

**Comment by tanyo520** [comment]
Is there any timeline to release Elsa 3.0 stable version ?
